src/services/providers/provider-sources/load-balance-provider.ts

```diff
@@ -79,40 +95,60 @@ function loadBalance(transports_: readonly Transport[], config: LoadBalanceProvi
         }
         throw noAvailableTransportsError ?? new Error('Failed to find a transport to execute the request'); // No transports available
       },
-      type: 'load-balance',
     });
   };
 }
 
-function initializeMetrics(): RPCMetrics {
-  return {
-    pending: 0,
-    samples: [],
-  };
-}
+class TransportInstance {
```

Comment:
I moved all request/metrics logic inside this class so that we could encapsulate the behavior and forget about it. It was getting hard to keep track of everything with multiple concurrent requests